### DISCUSSION TITLE
`brew audit` fails with "Can't find dependency '"rust"'." on Linux with Homebrew 4

### DISCUSSION THREAD

**Discussion by Tpt** [Homebrew/discussions#4233] [Tap maintenance and Homebrew development] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.0.1-60-g6ad9294
ORIGIN: https://github.com/Homebrew/brew
HEAD: 6ad92949e910041416d84a53966ec46b873e069f
Last commit: 7 hours ago
Core tap: N/A
HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew
HOMEBREW_BOOTSNAP: set
HOMEBREW_CASK_OPTS: []
HOMEBREW_CLEANUP_PERIODIC_FULL_DAYS: 3650
HOMEBREW_COLOR: set
HOMEBREW_CURL_PATH: /usr/bin/curl
HOMEBREW_DEVELOPER: set
HOMEBREW_FAIL_LOG_LINES: 150
HOMEBREW_GIT_EMAIL: 1589480+BrewTestBot@users.noreply.github.com
HOMEBREW_GIT_NAME: BrewTestBot
HOMEBREW_GIT_PATH: /usr/bin/git
HOMEBREW_MAKE_JOBS: 2
HOMEBREW_NO_AUTO_UPDATE: set
HOMEBREW_NO_EMOJI: set
HOMEBREW_NO_INSTALLED_DEPENDENTS_CHECK: set
Homebrew Ruby: 2.6.8 => /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8_1/bin/ruby
CPU: dual-core 64-bit skylake
Clang: 14.0.0
Git: 2.39.2 => /usr/bin/git
Curl: 7.81.0 => /usr/bin/curl
Kernel: Linux 5.15.0-1033-azure x86_64 GNU/Linux
OS: Ubuntu 22.04.1 LTS (jammy)
Host glibc: 2.35
/usr/bin/gcc: 11.3.0
/usr/bin/ruby: 3.0.2
glibc: N/A
gcc@11: N/A
gcc: N/A
xorg: N/A
```

### Output of `brew doctor`

```shell
All steps passed!
```

### Description of issue

With Homebrew 4, the `brew audit` step of [Oxigraph tap CI](https://github.com/oxigraph/homebrew-oxigraph/blob/master/.github/workflows/build.yml) fails with a ["Can't find dependency '"rust" error](https://github.com/oxigraph/homebrew-oxigraph/actions/runs/4210846959/jobs/7309021851). Here is [the formula code](https://github.com/oxigraph/homebrew-oxigraph/blob/master/Formula/oxigraph.rb).
[CI was working correctly on the last change a month ago](https://github.com/oxigraph/homebrew-oxigraph/actions/runs/3952296128/jobs/6767196526).

Sadly, I am not very experienced with Homebrew. Has Homebrew 4 made breaking change related to taps or Rust? I have not found anything relevant in the changelogs. Thank you so much for your help.

**Comment by carlocab** [comment, marked as answer]
Try adding `HOMEBREW_NO_INSTALL_FROM_API=1` to your workflow environment. `test-bot` should set this automatically, though... but give it a try and let me know if it helps.

**Reply by Tpt** [reply]
Thank you! [It is working now](https://github.com/oxigraph/homebrew-oxigraph/actions/runs/4211780291/jobs/7310348116).

**Reply by fopina** [reply]
I was struggling with this for a day, trying to understand why test-bot said there was no dependency `go` when locally it was fine...

```yaml
env:
  HOMEBREW_NO_INSTALL_FROM_API: 1
```
in the workflow did solve it!

If this is not being fixed in test-bot, can it at least be added to to the actions created by `brew tap-new` @carlocab ?
